docs/cookbook/server-managed-meta-tags.md

```diff
@@ -0,0 +1,351 @@
+
+
+### Client Side
+
+Copy the following code into your application. It should be rendered **once** in your application, such as in a [layout component
+](/guide/pages#creating-layouts).
+
+:::tabs key:frameworks
+== Vue
+
+```vue
+<script>
+import { Head } from '@inertiajs/vue3';
+import { usePage } from '@inertiajs/vue3';
+import { h } from 'vue';
+
+export default {
+  name: 'MetaTags',
+  setup() {
+    const page = usePage();
+
+    return () => {
+      const metaTags = page.props._inertia_meta || [];
+
+      return h(Head, {},
+        metaTags.map((meta) => {
+          const { tagName, innerContent, headKey, httpEquiv, ...attrs } = meta;
+
+          const attributes = {
+            key: headKey,
+            'head-key': headKey,
+            ...attrs
+          };
+
+          if (httpEquiv) {
+            attributes['http-equiv'] = httpEquiv;
+          }
+
+          let content = null;
+          if (innerContent != null) {
+            content = typeof innerContent === 'string'
+              ? innerContent
+              : JSON.stringify(innerContent);
+          }
+
+          return h(tagName, attributes, content);
+        })
+      );
+    };
+  }
+};
+</script>
+```
+
+== React
+
+```jsx
+import React from 'react';
+import { Head, usePage } from '@inertiajs/react';
+
+const MetaTags = () => {
+  const { _inertia_meta: meta } = usePage().props;
+  return (
+    <Head>
+      {meta.map((meta) => {
+        const { tagName, innerContent, headKey, httpEquiv, ...attrs } = meta;
+
+        let stringifiedInnerContent;
+        if (innerContent != null) {
+          stringifiedInnerContent =
+            typeof innerContent === 'string' ? innerContent : JSON.stringify(innerContent);
+        }
+
+        return React.createElement(tagName, {
+          key: headKey,
+          'head-key': headKey,
+          ...(httpEquiv ? { 'http-equiv': httpEquiv } : {}),
+          ...attrs,
+          ...(stringifiedInnerContent
+            ? { dangerouslySetInnerHTML: { __html: stringifiedInnerContent } }
+            : {}),
+        });
+      })}
+    </Head>
+  );
+};
+
+export default MetaTags;
+```
+
+== Svelte 4|Svelte 5
+
+```svelte
+<!-- MetaTags.svelte -->
+<script>
+  import { onMount } from 'svelte';
+  import { page } from '@inertiajs/svelte';
+
+  $: metaTags = ($page.props._inertia_meta ?? []).map(({
+    tagName,
+    headKey,
+    innerContent,
+    httpEquiv,
+    ...attrs
+  }) => ({
+    tagName,
+    headKey,
+    innerContent,
+    attrs: httpEquiv ? { ...attrs, 'http-equiv': httpEquiv } : attrs
+  }));
+
+  // Svelte throws warnings if we render void elements like meta with content
+  $: voidTags = metaTags.filter(tag => tag.innerContent == null);
+  $: contentTags = metaTags.filter(tag => tag.innerContent != null);
+
+  let ready = false;
+
+  onMount(() => {
+    // Clean up server-rendered tags
+    document.head
+      .querySelectorAll('[inertia]')
+      .forEach(el => el.remove());
```

Comment:
This is actually a lot simpler implementation than the React/Vue adapters have. Since you don't have to worry about built in `<Head />` components defining their own meta tags you can safely clear all the server rendered tags and re-render them from scratch.